src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/importer/price/PricesSpreadsheet.kt

```diff
@@ -58,20 +59,23 @@ class PricesSpreadsheet(
     val toLocation = locations[toLocationName]
       ?: throw RuntimeException("To location '$toLocationName' for supplier '$supplier' not found")
 
-    // Note: we do not need to worry about the effective year as we always start on a clean slate on import.
-    pricesRepository.findBySupplierAndFromLocationAndToLocation(supplier, fromLocation, toLocation)?.let {
-      throw RuntimeException("Duplicate price: '${fromLocation.siteName}' to '${toLocation.siteName}' for $supplier")
-    }
+    failIfPriceAlreadyExists(supplier, fromLocation, toLocation)
 
     return Price(
       supplier = supplier,
       fromLocation = fromLocation,
       toLocation = toLocation,
       priceInPence = price,
-      effectiveYear = 2020 // TODO remove this once we're not importing from spreadsheet anymore
+      effectiveYear = effectiveYear
     )
   }
 
+  private fun failIfPriceAlreadyExists(supplier: Supplier, fromLocation: Location, toLocation: Location) {
+    pricesRepository.findBySupplierAndFromLocationAndToLocationAndEffectiveYear(supplier, fromLocation, toLocation, effectiveYear)?.let {
```

Comment:
This is the key change here.  We need to check the new price against existing prices including the effective year.

Reply:
So is the effective year a column in the spreadsheet? Sorry, very out of the loop here.

Reply:
Hopefully my ramblings on Slack have helped here....